### DISCUSSION TITLE
Converter .IMG / .HEIC and .MOV to .PVT

### DISCUSSION THREAD

**Discussion by KovalevArtem** [RhetTbull/osxphotos#1576] [Ideas]
Is it possible to make a converter that combines photos and videos into one file (Live Photos)?
https://github.com/dariobanfi/convert2livephoto

I couldn't find enough information about the .PVT file extension.
https://docs.fileformat.com/image/pvt/
But if it really exists and these files are imported into Photos.app as Live Photos, then this seems to be a useful tool...

**Comment by RhetTbull** [comment]
I don't know anything about .pvt extension but I have created a tool for converting photos to live photos: [makelive](https://github.com/RhetTbull/makelive).  Also, if you import photos with `osxphotos import --auto-live` it will automatically convert matching photo+video pairs into live photos.

**Reply by JonLaliberte** [reply]
@RhetTbull Thanks for the work on this! 
I have a number of live photos that were uploaded to Google Photos, but when downloaded they are split into HEIC and MP4 files.

I was able to use makelive to add the metadata to them and importing those into Photos on macOS worked!

But, for some reason using the same files with osxphotos, like: `osxphotos import --auto-live IMG_0008.*` did not work, it imported the image and video separately.

It's not a huge deal though, I was able to just use makelive on them and drag/drop into Photos and it worked perfectly.

**Reply by RhetTbull** [reply]
@JonLaliberte if you have a pair of sample images you can send to OSXPhotos.py@gmail.com I'll take a look

**Reply by JonLaliberte** [reply]
@RhetTbull Sent just now. I tested a few others just now and noticed that the first 2 I tried didn't work, but then a few others I picked at random did work with osxphotos. But the example I sent imported as separate files with osxphotos, but makelive was able to create a pvt:

```
osxphotos import --auto-live IMG_5339.*
Collecting files to import... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Filtering import list for image & video files... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Grouping files by parent directory... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Grouping files into import groups... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
filepath1=PosixPath('/Users/jon/Downloads/Test-bak/IMG_5339.MP4'), filepath2=PosixPath('/Users/jon/Downloads/Test-bak/IMG_5339.HEIC')
Importing 2 files in 2 groups ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Done: imported 2 file groups, 0 errors
```

**Reply by RhetTbull** [reply]
I've figured out the issue. Sometimes (intermittently), when one of the files was an .mp4, it would be sorted in the wrong order for the "is this a possible live photo pair" check. I've fixed both the sort order and the code that does the check so it works regardless of what order the files are checked. I'll include a fix in the next release  (which may be a week or two as I'm in the middle of another fix for a different issue).

**Reply by JonLaliberte** [reply]
@RhetTbull Awesome, glad to hear it!
